plugins/chat/assets/javascripts/discourse/components/chat-footer.gjs

```diff
@@ -0,0 +1,83 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import i18n from "discourse-common/helpers/i18n";
+import eq from "truth-helpers/helpers/eq";
+
+export default class ChatFooter extends Component {
+  @service router;
+  @service chat;
+  @service chatApi;
+
+  @tracked threadsEnabled = false;
+
+  constructor() {
+    super(...arguments);
+    this.userThreadCount();
+  }
+
+  async userThreadCount() {
+    await this.chatApi.userThreadCount().then((result) => {
```

Comment:
generally speaking if you use await, don't use then. This is the pattern you should follow most of the cases:

```javascript
try {
  const result =  await this.chatApi.userThreadCount();
  this.threadsEnabled = result.thread_count > 0
} catch (error) {
  // do something with error
}
```